### PR TITLE
Cmm fork

### PR DESCRIPTION
**Pull request by dugdigg**
On our fork.

### CODE REVIEW

File: api/removeHost.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+$name = $_REQUEST['host'];
+$api_app = $_REQUEST['apiapp'];
+$api_token = $_REQUEST['apitoken'];
+
+# Check for bad characters in $name
+if (strlen($name) < 1) {
+  print "ERROR!! no host specified\n";
+  exit();
+}
+if (preg_match('/\ |\;|\'|&/', $name)) {
```
Comment (skpy) [inline api/removeHost.php]: Do we need to catch underscores and double quotes, too?

File: api/removeHost.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+# Check the api_app name and token passed to us
+$validate = ValidateToken($api_app, $api_token);
+if ($validate !== true) {
+  print "$validate";
+  exit();
+}
+
+# Set up a db connection
+$data = new mysqli("localhost", $db['user'], $db['pass'], $db['name']);
```
Comment (skpy) [inline api/removeHost.php]: Should `localhost` be replaced with `$db['host']` here?

File: api/getIPs.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+$api_token = $_REQUEST['apitoken'];
+
+# Check the api_app name and token passed to us
+$validate = ValidateToken($api_app, $api_token);
+if ($validate !== true) {
+  print "$validate";
+  exit();
+}
+
+$data = new mysqli("localhost", $db['user'], $db['pass'], $db['name']);
```
Comment (skpy) [inline api/getIPs.php]: Should `localhost` be replaced with `$db['host']` here?

**Comment by skpy** [comment]
In `getFreeIP.php` there is a prepared statement for querying SQL ([line 52](https://github.com/covermymeds/phpipam/blob/cmm_fork/api/getFreeIP.php#L52)), but the remainder of the SQL calls are direct queries.  In general, prepared statements offer protection against SQL injection attacks and I tend to prefer them.

Maybe @cjestel wants to weigh in here?

**Comment by cjestel** [comment]
Since they are already in use I would say lets stick with them everywhere since we already have a dependency on them. As @skpy mentioned, they are good for covering us on sql injection. They do introduce more complexity in dealing with poolers if we ever had a need for that, but I highly doubt we would in this app.

**Comment by skpy** [comment]
This looks okay for the short term.  The faster we can switch to using the stock upstream code and API, the happier I'll be.  😄

**Pull request merged**